library/src/main/java/com/getstream/sdk/chat/ImageLoader.kt

```diff
@@ -33,6 +41,43 @@ internal object ImageLoader {
             }
     }
 
+    fun ImageView.loadWithGifSupport(
+        uri: String?,
+        @DrawableRes placeholderResId: Int?,
+        onStart: () -> Unit = {},
+        onCancel: () -> Unit = {},
+        onError: (throwable: Throwable) -> Unit = {},
+        onSuccess: () -> Unit = {},
```

Comment:
Nit: We could wrap all of this up in an `onComplete`, if we don't care about the details at the call site for now.

Reply:
You're right. I've wanted to follow Coil API but it doesn't make sense for now